framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts

```diff
@@ -0,0 +1,450 @@
+import { codec } from '@liskhq/lisk-codec';
+import { hash, getRandomBytes } from '@liskhq/lisk-cryptography';
+import { InMemoryKVStore, KVStore } from '@liskhq/lisk-db';
+import * as testing from '../../../../../src/testing';
+import { UnlockCommand } from '../../../../../src/modules/dpos_v2/commands/unlock';
+import {
+	SELF_VOTE_PUNISH_TIME,
+	VOTER_PUNISH_TIME,
+	WAIT_TIME_SELF_VOTE,
+	WAIT_TIME_VOTE,
+	MODULE_ID_DPOS,
+	STORE_PREFIX_DELEGATE,
+	STORE_PREFIX_VOTER,
+	COMMAND_ID_UNLOCK,
+} from '../../../../../src/modules/dpos_v2/constants';
+import { delegateStoreSchema, voterStoreSchema } from '../../../../../src/modules/dpos_v2/schemas';
+import { TokenAPI, UnlockingObject, VoterData } from '../../../../../src/modules/dpos_v2/types';
+import { CommandExecuteContext } from '../../../../../src/node/state_machine/types';
+import { liskToBeddows } from '../../../../utils/assets';
+
+describe('UnlockCommand', () => {
+	let unlockCommand: UnlockCommand;
+	let db: KVStore;
+	let stateStore: StateStore;
+	let delegateSubstore: StateStore;
+	let voterSubstore: StateStore;
+	let mockTokenAPI: TokenAPI;
+	let lastBlockHeight: number;
+	let header: BlockHeader;
+	let unlockableObject: UnlockingObject;
+	let unlockableObject2: UnlockingObject;
+	let unlockableObject3: UnlockingObject;
+	let nonUnlockableObject: UnlockingObject;
+	let context: CommandExecuteContext;
+	let storedData: VoterData;
+	const delegate1 = { name: 'delegate1', address: getRandomBytes(32), amount: liskToBeddows(100) };
+	const delegate2 = { name: 'delegate2', address: getRandomBytes(32), amount: liskToBeddows(200) };
+	const delegate3 = { name: 'delegate3', address: getRandomBytes(32), amount: liskToBeddows(300) };
+	const delegate4 = { name: 'delegate4', address: getRandomBytes(32), amount: liskToBeddows(400) };
+	const defaultDelegateInfo = {
+		totalVotesReceived: BigInt(100000000),
+		selfVotes: BigInt(0),
+		lastGeneratedHeight: 0,
+		isBanned: false,
+		pomHeights: [],
+		consecutiveMissedBlocks: 0,
+	};
+	const publicKey = getRandomBytes(32);
+	const transaction = new Transaction({
+		moduleID: MODULE_ID_DPOS,
+		commandID: COMMAND_ID_UNLOCK,
+		senderPublicKey: publicKey,
+		nonce: BigInt(0),
+		fee: BigInt(100000000),
+		params: Buffer.alloc(0),
+		signatures: [publicKey],
+	});
+	const networkIdentifier = Buffer.from(
+		'e48feb88db5b5cf5ad71d93cdcd1d879b6d5ed187a36b0002cc34e0ef9883255',
+		'hex',
+	);
+
+	beforeEach(() => {
+		unlockCommand = new UnlockCommand(MODULE_ID_DPOS);
+		mockTokenAPI = {
+			unlock: jest.fn(),
+			lock: jest.fn(),
+		};
+		unlockCommand.addDependencies({
+			tokenIDDPoS: { chainID: 0, localID: 0 },
+			tokenAPI: mockTokenAPI,
+		});
+		db = new InMemoryKVStore() as never;
+		stateStore = new StateStore(db);
+		delegateSubstore = stateStore.getStore(MODULE_ID_DPOS, STORE_PREFIX_DELEGATE);
+		voterSubstore = stateStore.getStore(MODULE_ID_DPOS, STORE_PREFIX_VOTER);
+		lastBlockHeight = 8760000;
```

Comment:
```suggestion
		blockHeight = 8760000;
```
this height is currently being executed